Src/LiquidProjections.RavenDB/RavenProjector.cs

```diff
@@ -10,100 +11,36 @@ public class RavenProjector<TProjection> where TProjection : class, IHaveIdentit
     {
         private readonly Func<IAsyncDocumentSession> sessionFactory;
         private readonly int batchSize;
-        private readonly IProjectionCache<TProjection> cache;
-        private IEventMap<RavenProjectionContext> map;
+        private readonly RavenEventMapConfigurator<TProjection> mapConfigurator;
+        private readonly string stateKey;
 
         public RavenProjector(
             Func<IAsyncDocumentSession> sessionFactory,
-            IEventMapBuilder<TProjection, string, RavenProjectionContext> eventMapBuilder,
+            IEventMapBuilder<TProjection, string, RavenProjectionContext> mapBuilder,
+            IEnumerable<IRavenChildProjector> children = null,
             int batchSize = 1,
-            IProjectionCache<TProjection> cache = null)
-        {
-            this.sessionFactory = sessionFactory;
-            this.batchSize = batchSize;
-            this.cache = cache ?? new PassthroughCache<TProjection>();
-
-            CollectionName = typeof(TProjection).Name;
-
-            SetupHandlers(eventMapBuilder);
-        }
-
-        /// <summary>
-        /// The name of the collection under which the projections must be created (default: the name of the type). 
-        /// </summary>
-        public string CollectionName { get; set; }
-
-        private void SetupHandlers(IEventMapBuilder<TProjection, string, RavenProjectionContext> eventMapBuilder)
+            IProjectionCache cache = null,
+            string collectionName = null)
         {
-            eventMapBuilder.HandleCreatesAs(async (key, context, projector) =>
+            if (sessionFactory == null)

```

Comment:
I don't think it makes any sense to have specs for argument exceptions. They should be treated just like a part of the contracts of the methods. Those exceptions should never happen in correct code.